src/lib/consensus/proof_of_stake.ml

```diff
@@ -789,16 +789,60 @@ module Vrf = struct
         None )
 end
 
+module Lock_checkpoint = struct
```

Comment:
`Lock_checkpoint` is an oddly specific name for this module considering that it is only used for the `Curr` lock checkpoint. Perhaps a more generic name would be more appropriate, such as `Optional_state_hash`?